engine/lib/elgglib.php

```diff
@@ -1551,7 +1551,11 @@ function _elgg_ajax_page_handler($page) {
 				break;
 		}
 
-		echo elgg_view($view, $vars);
+		$output = elgg_view($view, $vars);
+
+		$output = _elgg_services()->deferredViews->resolveAll($output);
```

Comment:
A big challenge here is making sure this gets called for all output, including ajax not sent through standard channels like the ajax view handler.